mara_db/views.py

```diff
@@ -23,11 +23,44 @@ def navigation_entry():
                 description=f'The schema of the {alias} db',
                 uri_fn=lambda current_db=alias: flask.url_for('mara_db.index_page', db_alias=current_db))
             for alias, db in config.databases().items()
-            if (isinstance(db, dbs.PostgreSQLDB) and not isinstance(db, dbs.RedshiftDB))
-               or isinstance(db, dbs.MysqlDB)  # for now, only show postgres and mysql schemas
+            if supports_extract_schema(db)
         ])
 
 
+@singledispatch
+def supports_extract_schema(db: object) -> [bool]:
+    """
+    Returns true when the db supports schema extraction
+
+    Args:
+        db: The database which shall be tested for schema extraction
+    """
+    return False
+
+
+@supports_extract_schema.register(str)
+def __(alias: str):
+    return supports_extract_schema(dbs.db(alias))
+
+
+@supports_extract_schema.register(dbs.PostgreSQLDB)
+def __(db: dbs.PostgreSQLDB):
+    return not isinstance(db, dbs.RedshiftDB)
+
+
+@supports_extract_schema.register(dbs.MysqlDB)
+def __(db: dbs.MysqlDB):
+    return True
+
+
+@supports_extract_schema.register(dbs.SQLServerDB)
+def __(db: dbs.SQLServerDB):
+    # check if module pyodbc can be imported
+    import importlib
+    pyodbc_spec = importlib.util.find_spec("pyodbc")
+    return pyodbc_spec is not None
```

Comment:
```python
try:
    import pyodbc
    return True
except:
    return False
```

?

Reply:
well, yes, that was my first idea. But normally try/except clauses are slower in development langues (I am not sure here about python), so I came up with the `importlib` module idea;

your code is easier to read for sure

Reply:
shouldn't matter... Python code usually uses the 'try and handle failure' and not 'check if it can work and only then try it'.